IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/TestBuilder.cs

```diff
@@ -278,28 +272,27 @@ private object[] GatherTestBlockArguments(IServiceScope scope, MethodInfo execut
                 {
                     for (int i = 0; i < executeArgs.Length; i++)
                     {
-                        // Might be worth changing this to GetRequiredService and wrapping in a try/catch instead of checking if foundResult is null
-                        object foundResult =
-                            TestBlockResults.FirstOrDefault(tbr => tbr.GetType() == executeParams[i].ParameterType)
+                        object foundResult = TestBlockResults.FirstOrDefault(tbr => tbr.GetType() == executeParams[i].ParameterType)
                             ?? scope.ServiceProvider.GetService(executeParams[i].ParameterType);
-                        if (foundResult == null)
+                        if(foundResult == null)
                         {
-                            TestBlockException = new InvalidOperationException("Unable to resolve Execute method arguments");
+                            TestBlockException = new InvalidOperationException($"Unable to find an object or service for Execute parameter {executeParams[i]} on test block {tb.TestBlockType.FullName}.");
```

Comment:
```suggestion
                            TestBlockException = new InvalidOperationException($"Unable to find an object or service for Execute parameter {executeParams[i].ParameterType} on test block {tb.TestBlockType.FullName}.");
```

Reply:
@Keboo I was lazily trying to get both the parameter name and type. :) Is it worth separating these out to something like {executeParams[i].Name} and {executeParams[i].ParameterType.FullName}?

Reply:
O hm.... yea that works, but in general I prefer to be more explicit.

Reply:
Changed both to be more explicit.